packages/playground/src/components/deployment_data_dialog.vue

```diff
@@ -201,27 +201,34 @@ async function getGrafanaUrl() {
 getGrafanaUrl();
 
 async function getGPUInfo() {
-  showGpuCard.value = true;
   loadingCard.value = true;
 
   const grid = await getGrid(profileManager.profile!);
   if (grid) {
-    const gpuCards = await grid.zos.getNodeGPUInfo({ nodeId: contract.value.nodeId });
+    const nodeId = await grid.zos.capacity.getNodeIdFromContractId(
+      contract.value.contractId,
+      profileManager.profile!.mnemonic,
+    );
+
+    const gpuCards = await grid.zos.getNodeGPUInfo({ nodeId });
     const usedCards = gpuCards?.filter((card: any) => card.contract == contract.value.contractId);
 
     const cardsInfo = [];
     for (let i = 0; i < usedCards?.length; i++) {
       cardsInfo.push(getCardName(usedCards[i]));
     }
-    gpuInfo.value = cardsInfo.join(", ");
+    if (cardsInfo) {
+      gpuInfo.value = cardsInfo.join(", ");
+      showGpuCard.value = true;
+    }
   }
 
   if (gpuInfo.value == "") {
     showGpuCard.value = false;
   }
   loadingCard.value = false;
 }
-getGPUInfo();
+contract.value.metadata?.includes("fullvm") ? getGPUInfo() : undefined;
```

Comment:
__Suggestion__: this line is functional and will work, but i believe it's not the most clear or common way to structure this kind of logic. there are more readable alternatives that can improve code readability and maintainability
for example: 
`if(contract.value.metadata?.includes("fullvm"))  getGPUInfo();`